demos/python_demos/formula_recognition_demo/README.md

```diff
@@ -25,14 +26,69 @@ Second model is Decoder that takes as input:
 2. For each image, encoder extracts features from the image
 3. While length of the current formula is less then `--max_formula_len` or current token is not `END_TOKEN` Decode Step produces new tokens.
-5. The demo prints the decoded text into the console or in a file if `-o` parameter specified.
+4. The demo prints the decoded text to a file if `-o` parameter specified or into the console and (optionally) renders predicted formula into image.
+
+#### Rendering of the LaTeX formula into image
+User has an option to render the latex formula predicted by the demo application into an image.
+Regardless of what mode is selected (interactive or non-interactive) the process of the rendering of the formula is the same.
+##### Requirements for rendering
+Sympy python package is used for rendering. To install it, please, run:
+`pip install -r requirements.txt`
+Sympy package needs latex system installed in the operating system.
+For Windows you can use MiKTeX (just download and install it), for Ubuntu/MacOS you can use TeX Live:
+Ubuntu:
+`apt-get update && apt-get install texlive`
+MacOS:
+`brew install texlive`
+> Note: Other latex systems should also work.
+
+
+
+### Interactive mode
+The example of the interface:
+![](./interactive_interface.png)
+When User runs demo application with the `-i` option and passes `.mp4` video or number of the web-camera device as an argument (typically 0), window with the image simillar to above should pop up.
+
+Example of usage of the interactive mode:
+```
+python formula_recognition_demo.py <required args> -i 0
+```
+or
+```
+python formula_recognition_demo.py <required args> -i input_video.mp4
+```
+
+
+The window has four main sections:
+1. A red rectangle is placed on the center of this window. This is input "target", with the help of which User, moving the camera, can capture formula.
+2. Image from the input window will be binarized, preprocessed and fed to the network. Preprocessed and binarized image is placed on the top of the window (near `Model input` label)
+3. If the formula will be predicted with sufficient confidence score, it will be placed right under preprocessed image (near `Predicted` label)
+4. If rendering is available (see the previous Paragraph for details) and predicted formula does not contain latex grammar errors, it will be rendered and placed near `Rendered` label.
+
+Navigation keys:
+  * Use `q` button to quit from program
+  * Use `o` to decrease the size of the input (red) window
+  * Use `p` to increase the size of the input window
+
+The model inference process on the image is simillar to the Non-interactive mode with the exception that model infers asynchronously.
+Demo application works asynchronously. That means that model inference is performed in asynchronous mode and does not block main thread, so the image from the web-camera can move smoothly enough.
```

Comment:
There's a lot of repetition here:

* "model infers asynchronously"
* "demo application works asynchronously"
* "model inference is performed in asynchronous mode"

Couldn't it be simplified?

Reply:
Paraphrased these sentences